refinery_core/src/util.rs

```diff
@@ -58,9 +59,34 @@ pub fn find_migration_files(
     Ok(file_paths)
 }
 
+/// Loads SQL migrations from a path. This enables dynamic migration discovery, as opposed to
+/// embedding. The resulting collection is ordered by version.
+pub fn load_sql_migrations(location: impl AsRef<Path>) -> Result<Vec<Migration>, Error> {
+    let migration_files = find_migration_files(location, MigrationType::Sql)?;
+
+    let mut migrations = vec![];
+
+    for path in migration_files {
+        let sql = std::fs::read_to_string(path.as_path())
+            .map_err(|e| Error::new(Kind::InvalidMigrationPath(path.to_owned(), e), None))?;
```

Comment:
`fs::read_to_string` may return an error if:
> [It the contents of the file are not valid UTF-8, then an error will also be returned.](https://doc.rust-lang.org/std/fs/fn.read_to_string.html#errors)

We should probably introduce another variant `InvalidMigrationFile` for that

Reply:
Good point. Done.